src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -1,342 +1,369 @@
+                Assert.assertTrue(outputVC.hasAttribute(FOO_ID));
+                Assert.assertEquals(outputVC.getAttribute(FOO_ID), resourceVariant.getID());
+            } else {
+                Assert.assertFalse(outputVC.hasAttribute(FOO_ID));
+            }
+        }
 
-    @Test
-    public void testUsingExpression() throws IOException {
-        assertVariantContextsMatch(getTestFile("vcfexample3empty.vcf"),
-                new File(getToolTestDataDir() + "expected/testUsingExpression.vcf"),
-                Arrays.asList("--resource-allele-concordance",  "--resource:foo", getToolTestDataDir() + "targetAnnotations.vcf",
-                        "-G", "Standard", "-E", "foo.AF", "-L", getToolTestDataDir()+"vcfexample3empty.vcf"), b37_reference_20_21);
+        // test the --expression foo.AC annotation as an example of a resource INFO field that can be array-valued
+        final ArgumentsBuilder infoArgs = new ArgumentsBuilder(baseArgs).addArgument(VariantAnnotator.EXPRESSION_LONG_NAME, FOO_AC);
+        runCommandLine(infoArgs.getArgsList());
+        for (final VariantContext outputVC : VariantContextTestUtils.getVariantContexts(outputVCF)) {
+            final VariantContext resourceVariant = resourceVariantsByStartAndAlleles.get(keyForVariant(outputVC));
+            if (resourceVariant != null && resourceVariant.hasAttribute(AC)) {
+                Assert.assertTrue(outputVC.hasAttribute(FOO_AC));
+                Assert.assertEquals(outputVC.getAttribute(FOO_AC), resourceVariant.getAttribute(AC));
+            } else {
+                Assert.assertFalse(outputVC.hasAttribute(FOO_AC));
+            }
+        }
     }
 
     @Test
-    public void testUsingExpressionAlleleMisMatch() throws IOException {
-        assertVariantContextsMatch(getTestFile("vcfexample3empty-mod.vcf"),
-                new File(getToolTestDataDir() + "expected/testUsingExpressionAlleleMisMatch.vcf"),
-                Arrays.asList("--resource-allele-concordance",  "--resource:foo",  getToolTestDataDir() + "targetAnnotations.vcf",
-                        "-G", "Standard", "-E", "foo.AF", "-L", getToolTestDataDir()+"vcfexample3empty-mod.vcf"), b37_reference_20_21);
-    }
+    public void testNoReads() {
+        final File inputVCF = BASIC_INPUT;
+        final File outputVCF = createTempFile("output", ".vcf");
 
-    @Test
-    public void testPossibleDenovoAnnotation() throws IOException {
-        assertVariantContextsMatch(getTestFile("trioGGVCF.vcf.gz"),
-                new File(getToolTestDataDir() + "trioVA.denovo.vcf"),
-                Arrays.asList("-A", "PossibleDeNovo", "--" + StandardArgumentDefinitions.PEDIGREE_FILE_LONG_NAME, getToolTestDataDir() + "trio.ped"), b37_reference_20_21);
-    }
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .addVCF(inputVCF)
+                .addOutput(outputVCF)
+                .addInterval(new SimpleInterval("20", 1, 1));
```

Comment:
Can you add some annotations here?  It's not supposed to crash if you ask for annotations that require reads, but don't supply them, right?

Reply:
Done.  Added all annotations, just in case.

Reply:
Done. Added all annotations, just in case.